### PR TITLE
Add private `NonZero<T>` type alias.

### PR DESCRIPTION
**Pull request by reitermarkus**
According to step 2 suggested in https://github.com/rust-lang/rust/pull/100428#pullrequestreview-1767139731.

This adds a private type alias for `NonZero<T>` so that some parts of the code can already start using `NonZero<T>` syntax.

Using `NonZero<T>` for `convert` and other parts which implement `From` doesn't work while it is a type alias, since this results in conflicting implementations.

### CODE REVIEW

**Review by dtolnay — APPROVED** [review]
Thank you!

File: library/core/src/num/mod.rs [hunk 1]
```diff
@@ -791,7 +793,7 @@ impl u8 {
     #[rustc_const_stable(feature = "const_ascii_ctype_on_intrinsics", since = "1.47.0")]
     #[inline]
     pub const fn is_ascii_alphanumeric(&self) -> bool {
-        matches!(*self, b'0'..=b'9') | matches!(*self, b'A'..=b'Z') | matches!(*self, b'a'..=b'z')
+        matches!(*self, b'0'..=b'9' | b'A'..=b'Z' | b'a'..=b'z')
```
Comment (klensy) [inline library/core/src/num/mod.rs:796]: Why this unrelated change? this introduces branch https://rust.godbolt.org/z/8onesWshb

File: library/core/src/num/mod.rs [hunk 2]
```diff
@@ -1204,9 +1206,9 @@ impl u128 {
         swapped = "0x12907856341290785634129078563412",
         reversed = "0x48091e6a2c48091e6a2c48091e6a2c48",
         le_bytes = "[0x12, 0x90, 0x78, 0x56, 0x34, 0x12, 0x90, 0x78, \
-            0x56, 0x34, 0x12, 0x90, 0x78, 0x56, 0x34, 0x12]",
+      0x56, 0x34, 0x12, 0x90, 0x78, 0x56, 0x34, 0x12]",
         be_bytes = "[0x12, 0x34, 0x56, 0x78, 0x90, 0x12, 0x34, 0x56, \
-            0x78, 0x90, 0x12, 0x34, 0x56, 0x78, 0x90, 0x12]",
+      0x78, 0x90, 0x12, 0x34, 0x56, 0x78, 0x90, 0x12]",
```
Comment (klensy) [inline library/core/src/num/mod.rs:1211]: Broken format?
